data_steward/tools/copy_dataset_to_output_prod.py

```diff
@@ -0,0 +1,204 @@
+    """
+    User defined helper function to validate that the tier, deid_stage, release_tag parameter
+     follows the correct naming convention
+    :param tier: tier parameter passed through from either a list or command line argument
+    :param stage: deid_stage parameter passed through from either a list or command line argument
+    :param tag: release tag parameter passed through either the command line arguments
+    :return: None, breaks if not valid params are passed
+    """
+    validate_tier_param(tier)
+    validate_deid_stage_param(stage)
+    validate_release_tag_param(tag)
+
+
+def get_dataset_name(tier, release_tag, deid_stage):
+    """
+    Helper function to create the output prod dataset name based on the given criteria
+    This function should return a name for the final dataset only (not all steps along the way)
+    The function returns a string in the form: [C|R]{release_tag}[_base|_clean]
+
+    :param tier: controlled or registered tier intended for the output dataset
+    :param release_tag: release tag for dataset in the format of YYYYQ#R#
+    :param deid_stage: deid stage (deid, base or clean)
+    :return: a string for the dataset name
+    """
+    # validate parameters
+    validate_create_tier_args(tier, deid_stage, release_tag)
+
+    tier = tier[0].upper()
+    release_tag = release_tag.upper()
+    deid_stage = '' if release_tag == 'deid' else f'_{deid_stage}'
+
+    dataset_name = f"{tier}{release_tag}{deid_stage}"
+
+    return dataset_name
+
+
+def get_arg_parser() -> argparse.ArgumentParser:
+    """
+    Copy dataset from curation project to output-prod
+    """
+    argument_parser = argparse.ArgumentParser(description=__doc__)
+    argument_parser.add_argument(
+        '--run_as',
+        action='store',
+        dest='run_as_email',
+        help='Service account email address to impersonate',
+        required=True)
+    argument_parser.add_argument(
+        '-s',
+        '--src_project_id',
+        dest='src_project_id',
+        action='store',
+        help='Identifies the project containing the source dataset',
+        required=True)
+    argument_parser.add_argument('-o',
+                                 '--output_prod_project_id',
+                                 dest='output_prod_project_id',
+                                 action='store',
+                                 help='Identifies the output-prod project.',
+                                 required=True)
+    argument_parser.add_argument('-d',
+                                 '--src_dataset_id',
+                                 dest='src_dataset_id',
+                                 action='store',
+                                 help='The source dataset to copy.',
+                                 required=True)
+    argument_parser.add_argument(
+        '-r',
+        '--release_tag',
+        action='store',
+        dest='release_tag',
+        help='release tag for dataset in the format of YYYYQ#R#',
```

Comment:
You can also add `type=validate_release_tag_param` so it exits immediately if arg is not in the right format (can be unit tested)